### ISSUE TITLE
Abort in taint analysis demo

### ISSUE THREAD

**Issue by wideglide** [secure-software-engineering/phasar#245]
- [x] I have searched open and closed issues for duplicates
  - [x] I made sure that I am not using an old project version (DO: pull Phasar, update git submodules, rebuild the project and check if the bug is still there)

### Bug description
Attempting to reproduce the taint analysis example from the PhASAR-v0220-demo slides.

**Actual result:** 
```shell
root@9739619115ac:/usr/src/phasar# cat /data/test.c
#include <stdio.h>

int main() {
  char buffer[256];
  fgets(buffer, 256, stdin);
  fputs(buffer, stdout);
  return 0;
}
root@9739619115ac:/usr/src/phasar# /usr/local/llvm-10/bin/clang -g -o /data/test /data/test.c
root@9739619115ac:/usr/src/phasar# echo "Hello world" | /data/test
Hello world
root@9739619115ac:/usr/src/phasar# /usr/local/llvm-10/bin/clang -emit-llvm -S -g /data/test.c
root@9739619115ac:/usr/src/phasar# phasar-llvm -m test.ll -D ifds-taint
PhASAR v1220
A LLVM-based static analysis framework

Parsing JSON with source and sink functions
terminate called after throwing an instance of 'std::ios_base::failure[abi:cxx11]'
  what():   is not a valid path: iostream error
Aborted (core dumped)
````

### Context (Environment)

  - **phasar:**  \[a85e861e]
 
**Operating System:**
  - [x] Linux
  - [ ] Windows
  - [ ] macOS

**Build Type:** 
  - [x] Docker
  - [ ] custom build

**Comment by J03D03** [comment]
Hello @wideglide, 
thank you for reporting this issue. The `ifds-taint` analysis needs a source-and-sink-configuration. By default it should take [phasar-source-sink-function.json](https://github.com/secure-software-engineering/phasar/blob/master/config/phasar-source-sink-function.json) which doesn't seem to work. Therefore, you get an error while parsing the missing config file. 

For this example you could manually provide a `fgets-fputs-example.json` config file with the following content:
```
{
	"Source Functions": {
		"fgets": {
			"Args": [0],
			"Return": false
		}
	},
	"Sink Functions": {
		"fputs": {
			"Args": [0]
		}
	}
}
```

The command `phasar-llvm -m test.ll -D ifds-taint --analysis-config fgets-fputs-example.json` should then report a leak:
```
PhASAR v1220
A LLVM-based static analysis framework

Parsing JSON with source and sink functions
{
  "Sink Functions": {
    "fputs": {
      "Args": [
        0
      ]
    }
  },
  "Source Functions": {
    "fgets": {
      "Args": [
        0
      ],
      "Return": false
    }
  }
}
FOUND LEAK
No text report available!
```

Even without specifying `fputs` and `fgets` this should work since both
> are considered to as taint-sensitve functions by default

Reference:
https://github.com/secure-software-engineering/phasar/blob/c951bd91c3537b4220306f026f8027e2bb4c8495/include/phasar/PhasarLLVM/Utils/TaintConfiguration.h#L47-L68

But this does not seem to work

**Comment by MMory** [comment]
Hi @wideglide, is your issue resolved by the aforementioned merged PR #315 ?
